.circleci/config.yml

```diff
@@ -315,12 +338,15 @@ jobs:
       - install-dependencies:
           cache-id: hardhat-hackathon-boilerplate
           path: boilerplate
+      - fetch-latest-hardhat-release-tag
       - run:
           name: Update to the latest Hardhat release
           command: |
             # We can just use a release here because injection does not require rebuilding it.
             cd boilerplate/
-            npm update hardhat
+            # Install the latest release of Hardhat if the version matches the expected format.
+            [[ "${HARDHAT_LATEST_RELEASE_TAG}" =~ ^hardhat@([0-9]+\.){2}[0-9]+$ ]] && npm install ${HARDHAT_LATEST_RELEASE_TAG}
```

Comment:
ok, this is fine assuming it will fail, when the version does not match and not just continue without installing it.